### ISSUE TITLE
Fundamentally restructure the timetable

### ISSUE THREAD

**Issue by oliverjam** [foundersandcoders/coursebook#630]
WIP issue, will add more info later

We've had the same structure forever, so maybe it's time for a change. How about making Monday a work-from-home-do-the-workshops-yourself day, and Tuesday a discuss-workshops-all-together day?

**Comment by sofer** [comment]
This new approach would meet three objectives that we do not currently do very well:
- Allow learners to spend time on self-study
- Allow them more opportunity to master the material at their own speed
- Allow us to flip the classroom, i.e. come together to discuss material already studied and to provide remedial support.

It could also potentially reduce the number of required in-class days to one! 

I would be in favour of doing a trial week this term using some version of this approach, perhaps setting all the workshops on Friday afternoon and leaving them to it until Tuesday.

**Comment by sofer** [comment]
If we did it this term, the obvious week to go for would be Week 6. It's a bit of a cheat, because it's not quite like earlier weeks, but there is more-or-less nothing going on that week on Monday other than workshops, so we could just leave them to it.

We could also give the Week 1 workshops as homework over the holidays to FAC25, which would give us more time on Monday to do the Welcome talk and then to crack on with workshop reviews. If it goes well, we could adapt the approach to the following weeks, too.

**Comment by sofer** [comment]
Some feedback from a FAC24 learner:
- In general, having individual study time would suit them.
- Is one day enough for all workshops? Perhaps give over some time on Tuesday morning for finishing them off or doing the final one.
- Consider having Tuesday tutorial groups based on the project teams, so that they can support each other even before the projects kick off.
- It might be worth doing the project and spikes intro last thing on Friday, so that learners are thinking about the project and have picked their roles before they start on the workshops.

**Comment by Albadylic** [comment]
I was concerned the workshops might not fit into one day, but looking in detail at the schedules for weeks 1 -6, I'm now convinced they can.

| Week | Workshops | Spikes | Project time | Morning challenge | Total | 
| --- | --- | --- | --- | --- | --- |
| 1 | 8.5 | 1 | 11.5 | 1 | 22 |
|2 | 8.25 | 1 | 11 | 1 | 21.25 |
| 3 | 8.5 | 1 | 10 | 1 | 20.5 |
| 4 | 2.83...| 3.5 | 17 | 0 | 23.33...|
| 5 | 11.5 | 0 | 11 | 1 | 23.5 |
| 6| 6.25 | 0 | 17.25 | 0 | 23.5 |

_Times are in hours_

A couple of notes:
- Workshops are modally 8.5 hours, which is a bit longer than a work day, even if we consider that the discussion time is extracted from that total. Can we squeeze them to 7 hours of work? This may be moot, since everyone will take a different amount of time to complete them.
- Server-side (4) has longer tech spikes. These might not neatly fit into being workshops.
- Client-side week (5) has more workshops, we'd probably need to spill some of these into project time or cut some out.
- Full-stack week (6) has more project time than any other, and no spikes.

---

[A proposal for a new default schedule](https://deploy-preview-676--fac-coursebook.netlify.app/course/syllabus/apprenticeship/default/schedule/)

Some notes:
- Spikes get an extra 15 mins
- MC could move to Wednesday morning if we want more than 120 mins of discussion
- We sometimes have short additional talks (e.g. intro to software dev process). These could fit after spike presentations on Wednesday morning, or on Thursday morning.

**Comment by sofer** [comment]
Another aspect of the restructure is to think of the programme as having 8 3-hour slots, Tues-Fri, 10am-1pm and 2-5pm. This has the benefit that moving stuff around is less likely to twist the timetable out of shape and it may be easier to allocate mentors a bit more more efficiently to equal-length 3-hour slots. it doesn't mean that we close the classroom at 5, it just means that any activities after then are treated as separate, leaving space for talks or prep work or code reviews or other activities, pretty much as at present, but with a clearly-defined boundary at 5pm.

ps the subject of this issue could be "Fundamentally restructure the *timetable*", because we are not proposing any changes to course content.

**Comment by reubengt** [comment]
list the workshops in a separate tab (next to schedule/projects etc)

**Comment by sofer** [comment]
Following on from yesterday's comment, I have been looking at the timetable again and I find it a bit hard to follow. If there is any way we could simplify it, I would be very much happier. The danger with a schedule that changes from week to week is that it becomes more brittle and harder for each new facilitator to manage and adapt. That could become a problem.

**Comment by reubengt** [comment]
@sofer I believe gregor only updated the default schedule - so the individual weekly schedules on that link are broken and should not be looked at.

The default schedule doesn't change much in weeks 1-6 apart from slotting in things like web science and tfb

**Comment by oliverjam** [comment]
Some notes following our discussion today:

## Learning/workshops

- Would it make more sense to condense the learning into a single long resource?
   - We no longer have the limitation of needing to break things into 1/2 hour blocks
   - This would make maintenance easier: one starter repo for each topic (could make it easier to use GitHub Classroom)
- We might move the morning challenges to stretch goals for people who have additional time/understanding to push themselves

## Learning/workshops discussion

- ~3 hours for discussion—2 hours in small groups (probably of 3 learners each) and 1 hour as a whole cohort with the mentors.
- Small groups should be given the space to share struggles ask "stupid questions"—no mentors/CF present
- Small groups should have some direction: first hour discussing workshops/struggles/solutions, second hour on higher level questions provided by us

## Projects

- Tech spikes are not relevant enough to the projects right now
- Projects should have at least one (or a few options of) technical problems to solve beyond the week's "boilerplate"
- We might want to consider much more specific project ideas so people don't get lost at the creative step
- A better project structure might look like:
   1. Solve the technical problem (a spike with throwaway code)
   2. Come up with an idea/plan the project
   3. Build the actual features (should be simpler since the hard part is already solved)
   4. Respond to feedback (after code review)
   This is more similar to the structure of a real project, where you wouldn't just dive into building something from the start

**Comment by Albadylic** [comment]
## Learning/workshops discussion*

Morning challenge stays
80 minutes - groups of 3, discussing workshop and coming up with a high level concept / question to go into detail on
40 minutes - each group present their 'struggle' and discuss with the other members of the cohort and mentors

**Comment by Albadylic** [comment]
One unresolved issue is where to put the intro - at the end of the day on Friday feels a bit weird, but also makes the most sense.

**Comment by reubengt** [comment]
Introducing next week on Fridays - also check that people have things installed and set up for the workshops as well.

**Comment by reubengt** [comment]
week 4 has a slightly weird structure as well

**Comment by Albadylic** [comment]
I think we can close this - the timetable has been restructured. However, we'll need to shift everything along next term when we move to Mon - Thurs
